### DISCUSSION TITLE
Get all branches information which has github action enabled

### DISCUSSION THREAD

**Discussion by shreelola** [community/community#34477] [Actions]
Are there API endpoints to list all branches with GitHub action workflow enabled? 

We would like to list each repo and list of branches which has GitHub action enabled. 
If that's not available, is there any way to restrict GitHub actions for specific branches? For example, let's say the Github admin added my repository to run GitHub actions,   and I have write access to the specific repo. I can create a thousand runs, use all allocated resources, and bring down the system.  
How can we restrict such actions, and what is the best workflow to handle multiple repositories across the org.

**Comment by byrneh** [comment]
HI @shreelola actions are enabled at the reposiorty level.  There is an organization API that provides this information
[#get-github-actions-permissions-for-an-organization](https://docs.github.com/en/rest/actions/permissions#get-github-actions-permissions-for-an-organization)

**Reply by shreelola** [reply]
@byrneh Correct, we are trying to restrict GitHub actions to specific branches. I couldn't find any restriction policies for branches.
Example: Let's say I have access to the repository ABC and the GitHub action is enabled at the org level from my org. Since I have write access to that repository, I can misuse the environment and run GitHub actions for all my branches. If we can restrict github actions to certain branches or know branches that enabled GitHub actions we can avoid that.
